zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/JobControllerTest.java

```diff
@@ -7,280 +7,141 @@
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
 
-import com.jayway.jsonpath.JsonPath;
 import io.camunda.service.JobServices;
-import io.camunda.zeebe.broker.client.api.BrokerClient;
-import io.camunda.zeebe.broker.client.api.dto.BrokerRejection;
-import io.camunda.zeebe.broker.client.api.dto.BrokerRejectionResponse;
-import io.camunda.zeebe.gateway.api.job.ActivateJobsStub;
-import io.camunda.zeebe.gateway.api.util.StubbedBrokerClient;
-import io.camunda.zeebe.gateway.impl.broker.request.BrokerActivateJobsRequest;
-import io.camunda.zeebe.gateway.impl.job.ActivateJobsHandler;
-import io.camunda.zeebe.gateway.impl.job.RoundRobinActivateJobsHandler;
+import io.camunda.service.security.auth.Authentication;
 import io.camunda.zeebe.gateway.protocol.rest.JobActivationResponse;
-import io.camunda.zeebe.gateway.rest.ResponseMapper;
-import io.camunda.zeebe.gateway.rest.controller.util.ResettableJobActivationRequestResponseObserver;
-import io.camunda.zeebe.protocol.Protocol;
-import io.camunda.zeebe.protocol.record.RejectionType;
-import io.camunda.zeebe.protocol.record.intent.Intent;
-import io.camunda.zeebe.scheduler.Actor;
-import io.camunda.zeebe.scheduler.ActorScheduler;
-import io.camunda.zeebe.scheduler.clock.ActorClock;
-import io.camunda.zeebe.scheduler.clock.ControlledActorClock;
+import io.camunda.zeebe.protocol.impl.record.value.job.JobRecord;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.atomic.AtomicInteger;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
-import org.springframework.boot.test.context.TestConfiguration;
-import org.springframework.boot.test.mock.mockito.SpyBean;
-import org.springframework.context.annotation.Bean;
+import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.http.MediaType;
-import org.springframework.util.unit.DataSize;
 
 @WebMvcTest(JobController.class)
 public class JobControllerTest extends RestControllerTest {
 
   static final String JOBS_BASE_URL = "/v2/jobs";
 
-  @Autowired ActivateJobsHandler<JobActivationResponse> activateJobsHandler;
-  @Autowired StubbedBrokerClient stubbedBrokerClient;
-  @SpyBean ResettableJobActivationRequestResponseObserver responseObserver;
+  @MockBean JobServices<JobActivationResponse> jobServices;
+  @MockBean ResponseObserverProvider responseObserverProvider;
 
   @BeforeEach
   void setup() {
-    responseObserver.reset();
+    when(jobServices.withAuthentication(any(Authentication.class))).thenReturn(jobServices);
   }
 
   @Test
-  void shouldActivateJobsImmediatelyIfAvailable() throws Exception {
```

Comment:
ℹ️ Just for reference in case someone comes across this: the job activation tests will be added with this [JobControllerRoundRobinTest](https://github.com/camunda/camunda/pull/20089/files#diff-b172358d9f2fd6f395dfeb2ecf928264c077fe07da1ed5b9d681557a888bbd59) from https://github.com/camunda/camunda/pull/20089.